source/boundary_composition/ascii_data.cc

```diff
@@ -0,0 +1,428 @@
+                             "Time from which on the data file with number 'First data "
+                             "file number' is used as boundary condition. Previous to this "
+                             "time, a no-slip boundary condition is assumed. Depending on the setting "
+                             "of the global 'Use years in output instead of seconds' flag "
+                             "in the input file, this number is either interpreted as seconds or as years.");
+          prm.declare_entry ("First data file number", "0",
+                             Patterns::Integer (),
+                             "Number of the first data file to be loaded when the model time "
+                             "is larger than 'First data file model time'.");
+          prm.declare_entry ("Decreasing file order", "false",
+                             Patterns::Bool (),
+                             "In some cases the boundary files are not numbered in increasing "
+                             "but in decreasing order (e.g. 'Ma BP'). If this flag is set to "
+                             "'True' the plugin will first load the file with the number "
+                             "'First data file number' and decrease the file number during "
+                             "the model run.");
+          prm.declare_entry ("Scale factor", "1",
+                             Patterns::Double (0),
+                             "Scalar factor, which is applied to the boundary data. "
+                             "You might want to use this to scale the data.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Boundary composition model");
+      {
+        prm.enter_subsection("Ascii data model");
+        {
+          // Get the path to the data files. If it contains a reference
+          // to $ASPECT_SOURCE_DIR, replace it by what CMake has given us
+          // as a #define
+          data_directory    = prm.get ("Data directory");
+          {
+            const std::string      subst_text = "$ASPECT_SOURCE_DIR";
+            std::string::size_type position;
+            while (position = data_directory.find (subst_text),  position!=std::string::npos)
+              data_directory.replace (data_directory.begin()+position,
+                                      data_directory.begin()+position+subst_text.size(),
+                                      ASPECT_SOURCE_DIR);
+          }
+
+          data_file_name    = prm.get ("Data file name");
+
+          scale_factor      = prm.get_double ("Scale factor");
+          data_points[0]    = prm.get_double ("Number of x grid points");
+          data_points[1]    = prm.get_double ("Number of y grid points");
+          data_points[2]    = prm.get_double ("Number of z grid points");
```

Comment:
I think `data_points` should only be an array of size `dim`, and in that case you'd only read the last component if you're in 3d.